htmd/smallmol/chemlab/periodictable.py

```diff
@@ -0,0 +1,189 @@
+
+from rdkit.Chem.rdchem import HybridizationType, BondType, ChiralType
+from rdkit.Chem import GetPeriodicTable
+from rdkit import rdBase
+import numpy as np
+
+
+_heavy_atoms = ['C', 'N', 'O', 'F',
+                'Si','P', 'S', 'Cl', 'Br', 'I']
+
+_hetero_atoms = ['N', 'O', 'F',
+                'Si','P', 'S', 'Cl', 'Br', 'I']
+
+_hetero_atoms = ['N', 'O', 'F', 'P', 'S', 'Cl', 'Br', 'I']
+_halogen_atoms = ['F', 'Cl', 'Br', 'I']
+
+
+_hybridizations_IdxToType = HybridizationType.values
+_hybridizations_StringToType = {'S': HybridizationType.S,
+                                'SP': HybridizationType.SP,
+                                'SP2': HybridizationType.SP2,
+                                'SP3': HybridizationType.SP3}
+
+rdBase.DisableLog('rdApp.error')
+atom_mapping = {"Hydrophobe": 0,
+                "LumpedHydrophobe": 0,
+                "Aromatic": 1,
+                "Acceptor": 2,
+                "Donor": 3,
+                "PosIonizable": 4,
+                "NegIonizable": 5}
+
+_chiral_type = {ChiralType.CHI_TETRAHEDRAL_CW:'clockwise',
+                ChiralType.CHI_TETRAHEDRAL_CCW:'anitclockwise'}
+_chiral_type_Dict = ChiralType.values
+
+_bondtypes_IdxToType = BondType.values
+_bondtypes_StringToType = {'SINGLE': BondType.SINGLE,
+                           'DOUBLE': BondType.DOUBLE,
+                           'TRIPLE': BondType.TRIPLE}
+
+_atoms = {'H': [ 'H', 0, 0, '', 1, 1],
+         'C': ['C', 0, 0, '', 4, 1],
+        }
+
+class PeriodicTable:
+    """
+    Class that manages all the chemical information about atoms based on the atom. It exposes the rdkit PeriodicTable
+    functionalities.
+    """
+
+    def __init__(self):
+        self.PeriodicaTable = GetPeriodicTable()
+
+    def _evaluateMissingAtoms(self, valence, btypes):
+        """
+        Returns the difference bewtween the current valence computed and the expected one.
+
+        Parameters
+        ----------
+        valence: int
+            The valence of the atom
+        btypes: list
+            A list of the bondtypes connected to the atom
+
+        Returns
+        -------
+        missatoms: int
+            The number of atoms that miss for completing the atom valence
+
+        """
+        normbondvalence = sum([ 1.5 if int(b) == 12 else int(b) for b in btypes  ])
+
+        missatoms = int(valence - normbondvalence)
+
+        return missatoms
+
+    def getRadiusBond(self, element):
+        """
+        Returns the atom radius when bonded
+
+        Parameters
+        ----------
+        element: str
+            The element of the atom
+
+        Returns
+        -------
+        bradius: float
+            The radius of the bond when bonded
+
+        """
+
+        pT = self.PeriodicaTable
+        bradius = pT.GetRb0(element)
+
+        return bradius
+
+
+    def getMissingValence(self, atomidx, smallmol, formalcharge=0):
```

Comment:
I would switch the order of arguments to be `smallmol, atomidx` Because when you use the formal charge it looks messy having the object in the middle of the properties.

Reply:
I see your point. Modified as you suggested